### ISSUE TITLE
Creating hooks with class 

### ISSUE THREAD

**Issue by shriharip** [rrousselGit/flutter_hooks#212]
**Describe what scenario you think is uncovered by the existing examples/articles**
I am trying to create a Hook (storage) using the example of the class TimeAlive.

Here is the code.
````
void main() {
  runApp(
    MaterialApp(
      title: 'Reading and Writing Files',
      home: FlutterDemo(),
    ),
  );
}

 useStorage() {
  return use(_Storage());
}

class _Storage extends Hook{
  const _Storage();

  @override
  _StorageState createState() => _StorageState();
}

class _StorageState extends HookState {
  int counter;
  
  @override
  void initHook() {
    super.initHook();
    readCounter().then((int value) {
      setState(() {
        counter = value;
      });
    });
  }

    Future<String> get _localPath async {
    final directory = await getApplicationDocumentsDirectory();
     return directory.path;
  }

  Future<File> get _localFile async {
    final path = await _localPath;
    return File('$path/counter.txt');
  }

  Future<int> readCounter() async {
    try {
      final file = await _localFile;

      // Read the file
      String contents = await file.readAsString();
      Map<String, dynamic> dbval = json.decode(contents);

      return dbval['counter'];
      // return int.parse(contents);
    } catch (e) {
      // If encountering an error, return 0
      return 0;
    }
  }

  Future<File> writeCounter(String db) async {
    final file = await _localFile;

    // Write the file
    return file.writeAsString(db);
  }


  @override
  void build(BuildContext context) {}

  @override
  void dispose() {
    super.dispose();
  }
}

class FlutterDemo extends HookWidget {
 
  @override
  Widget build(BuildContext context) {
    var st = useStorage();
    final counter = useState(0);
   print(st) //always null; 
return Scaffold(
      appBar: AppBar(title: Text('Reading and Writing Files')),
      body: Center(
        child: Text(
          'Button tapped ${counter.value} time${counter.value == 1 ? '' : 's'}.',
        ),
      ),
      floatingActionButton: FloatingActionButton(
        onPressed: (){counter.value++;},
        tooltip: 'Increment',
        child: Icon(Icons.add),
      ),
    );
  }
}
````

**Describe why existing examples/articles do not cover this case**
I saw both the animationController and TimeAlive example and adapted to it. 

But the value of st. using usestorage is always null..

**Additional context**
using latest hooks package and flutter 
Flutter 1.22.4 • channel stable • https://github.com/flutter/flutter.git
Framework • revision 1aafb3a8b9 (2 months ago) • 2020-11-13 09:59:28 -0800
Engine • revision 2c956a31c0
Tools • Dart 2.10.4

**Comment by rrousselGit** [comment]
See `useState` for a proper example of how to return a value

TL;DR, you need to return the value in `build`

**Comment by shriharip** [comment]
Thanks @rrousselGit . Got it
